source/common/config/filesystem_subscription_impl.h

```diff
@@ -33,8 +36,10 @@ class FilesystemSubscriptionImpl : public Config::Subscription<ResourceType> {
     callbacks_ = &callbacks;
     watcher_->addWatch(path_, Filesystem::Watcher::Events::MovedTo, [this](uint32_t events) {
       UNREFERENCED_PARAMETER(events);
-      refresh();
+      refresh(false);
     });
+    // Attempt to read, gracefully fail if the file isn't there yet.
```

Comment:
What's the reasoning behind differentiating this case? Seems like it adds code complexity for not too much benefit?